src/main/java/de/chojo/repbot/ReputationBot.java

```diff
@@ -237,14 +238,27 @@ private void initLocalization() {
     }
 
     private void initBot() {
+        metrics = new Metrics(dataSource);
         RestAction.setDefaultFailure(throwable -> {
             if (throwable instanceof InsufficientPermissionException) {
                 PermissionErrorHandler.handle((InsufficientPermissionException) throwable, shardManager, localizer, configuration);
                 return;
             }
+            if (throwable instanceof ErrorResponseException e) {
+                if (e.getErrorResponse() == ErrorResponse.UNKNOWN_INTERACTION) {
+                    metrics.service().failedInteraction();
+                    log.debug("Interaction timed out", e);
+                    return;
+                }
+            }
             log.error(LogNotify.NOTIFY_ADMIN, "Unhandled exception occured: ", throwable);
         });
-        metrics = new Metrics(dataSource);
+
+        RestAction.setDefaultSuccess(suc -> {
+            if (suc instanceof InteractionHook) {
+                metrics.service().successfulInteraction();
+            }
+        });
```

Comment:
As mentioned above, we can't be sure that this captures all interactions because this handler only gets called when we have not specified an own handler for the queue() call.